warehouse/query-core/src/main/java/datawave/query/iterator/logic/ContentSummaryIterator.java

```diff
@@ -0,0 +1,329 @@
+    }
+
+    @Override
+    public void next() throws IOException {
+        tk = null;
+        tv = null;
+
+        if (log.isTraceEnabled()) {
+            log.trace("{} calling next on {}", source.hasTop(), scanRange);
+        }
+
+        // find a valid dt/uid (depends on initial column families set in seek call)
+        String dtUid = null;
+        while (source.hasTop() && dtUid == null) {
+            Key top = source.getTopKey();
+            String thisDtUid = getDtUidFromDKey(top);
+            // if this dt and uid are in the accepted column families...
+            if (columnFamilies.contains(thisDtUid) == inclusive) {
+                // we can use this document
+                dtUid = thisDtUid;
+            } else {
+                seekToNextUid(top.getRow(), thisDtUid);
+            }
+        }
+
+        // if no more d keys, then we are done.
+        if (!source.hasTop() || dtUid == null) {
+            return;
+        }
+
+        Key top = source.getTopKey();
+
+        // this is where we will save all the content found for this document
+        final Map<String,byte[]> foundContent = new HashMap<>();
+
+        // while we have d keys for the same document
+        while (source.hasTop() && dtUid.equals(getDtUidFromDKey(source.getTopKey()))) {
+            top = source.getTopKey();
+
+            // get the view name
+            String currentViewName = getViewName(top);
+
+            for (String name : viewSummaryOrder) {
+                if (name.endsWith("*")) {
+                    name = name.substring(0, name.length() - 1);
+                    if (currentViewName.startsWith(name)) {
+                        foundContent.put(currentViewName, source.getTopValue().get());
+                    }
+                } else {
+                    if (currentViewName.equalsIgnoreCase(name)) {
+                        foundContent.put(currentViewName, source.getTopValue().get());
+                    }
+                }
+            }
+
+            // get the next d key
+            source.next();
+        }
+
+        // create the summary
+        String summary = new SummaryCreator(viewSummaryOrder, foundContent, summarySize).createSummary();
+        if (summary != null) {
+            tk = new Key(top.getRow(), new Text(dtUid), new Text(summary), top.getColumnVisibility());
+            tv = new Value();
+            return;
+        }
+
+        // If we get here, we have not found content to summarize, so return null
+        tk = null;
+        tv = null;
+    }
+
+    /**
+     * Seek to the dt/uid following the one passed in
+     *
+     * @param row
+     *            a row
+     * @param dtAndUid
+     *            the dt and uid string
+     * @throws IOException
+     *             for issues with read/write
+     */
+    private void seekToNextUid(Text row, String dtAndUid) throws IOException {
+        Key startKey = new Key(row, Constants.D_COLUMN_FAMILY, new Text(dtAndUid + '.'));
```

Comment:
is a '.' the right character here? the dtAndUid will be followed by a null-byte right?

Reply:
done